### PR TITLE
Implement StateStore for fetching accounts and transactions and saving updated accounts in the database - Closes #2820

### LINKED ISSUE #2820: Implement StateStore for fetching accounts and transactions and saving updated accounts in the database
### Expected behavior
In order to verify/apply transactions in memory, we need to implement StateStore. StateStore manages accounts and transactions in memory. Furthermore, it allows the functionality to update accounts in memory and save the updated state in the database on completion of processing of block. 
### Actual behavior
StateStore currently is not implemented.

### Which version(s) does this affect? (Environment, OS, etc...)
1.6

### PR DESCRIPTION
**Pull request by SargeKhan**
### What was the problem?
We need to implement StateStore for transactions and accounts as part of transaction processing efficiency LIP
### How did I fix it?
Added functionality for StateStore
### How to test it?
Run tests
### Review checklist

* The PR resolves #2820
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/logic/state_store/account_store.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		if (!element) {
+			throw new Error(
+				`${this.name} with ${this.primaryKey} = ${primaryValue} does not exist`
+			);
+		}
+		return element;
+	}
+
+	getOrDefault(primaryValue) {
+		let element = this.data.find(
```
Comment (shuse2) [inline framework/src/modules/chain/logic/state_store/account_store.js]: prefer to use `const`

File: framework/src/modules/chain/logic/state_store/account_store.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		this.data = [];
+		this.updatedKeys = {};
+		this.primaryKey = 'address';
+		this.name = 'Account';
+		this.mutate = mutate;
+		this.originalData = [];
+		this.originalUpdatedKeys = {};
+	}
+
+	async cache(filter, tx) {
```
Comment (shuse2) [inline framework/src/modules/chain/logic/state_store/account_store.js]: we don't have access to `tx` in the transaction module ='(

File: framework/src/modules/chain/logic/state_store/transaction_store.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+class TransactionStore {
+	constructor(transactionEntity) {
+		this.transaction = transactionEntity;
+		this.data = [];
+		this.primaryKey = 'id';
+		this.name = 'Transaction';
+	}
+
+	async cache(filter, tx) {
```
Comment (shuse2) [inline framework/src/modules/chain/logic/state_store/transaction_store.js]: same as above

File: framework/src/modules/chain/logic/state_store/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const TransactionStore = require('./transaction_store');
+
+class StateStoreManager {
+	constructor(storage, cb) {
+		this.entities = {
+			Account: storage.entities.Account,
+			Transaction: storage.entities.Transaction,
+		};
+
+		this.sandbox = {};
```
Comment (shuse2) [inline framework/src/modules/chain/logic/state_store/index.js]: is this `sandbox` right terminology? maybe just `stateStore`?

File: framework/src/modules/chain/logic/state_store/account_store.js (hunk 1 above)
Comment (SargeKhan) [reply framework/src/modules/chain/logic/state_store/account_store.js]: We initalize `element` variable if it doesn't exist. That's why I am using const.

Comment (shuse2) [reply framework/src/modules/chain/logic/state_store/account_store.js]: maybe something like
```
const element = this.data.find(xxx);
if (element)
  return element;

const defaultElement = { ...defaultAccount, xxx};
this.data.push(defaultElement)
return defaultElement;
```

File: framework/src/modules/chain/logic/state_store/account_store.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	u_nameExist: false,
+	multiMin: 0,
+	u_multiMin: 0,
+	multiLifetime: 0,
+	u_multiLifetime: 0,
+};
+
+class AccountStore {
+	constructor(accountEntity, { mutate } = { mutate: true }) {
+		this.account = accountEntity;
```
Comment (pablitovicente) [inline framework/src/modules/chain/logic/state_store/account_store.js]: wouldn't it be better to name this property `accountEntity` so when reading other methods it's very clear that the operation it's an storage access?

File: framework/src/modules/chain/logic/state_store/index.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	createStore(options) {
+		this.stateStore = {
+			account: new AccountStore(this.entities.Account, options),
+			transaction: new TransactionStore(this.entities.Transaction, options),
+		};
+		return this.stateStore;
+	}
+
+	createSnapshot() {
+		this.sandbox.account.createSnapshot();
```
Comment (pablitovicente) [inline framework/src/modules/chain/logic/state_store/index.js]: Need to update to this.stateStore maybe?

File: framework/src/modules/chain/logic/state_store/index.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		this.sandbox.transaction.createSnapshot();
```
Comment (pablitovicente) [inline framework/src/modules/chain/logic/state_store/index.js]: Need to update to this.stateStore maybe?

File: framework/src/modules/chain/logic/state_store/index.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	restoreSnapshot() {
+		this.sandbox.account.restoreSnapshot();
+		this.sandbox.transaction.restoreSnapshot();
```
Comment (pablitovicente) [inline framework/src/modules/chain/logic/state_store/index.js]: Need to update to this.stateStore maybe?

File: framework/src/modules/chain/logic/state_store/account_store.js (hunk 5 above)
Comment (SargeKhan) [reply framework/src/modules/chain/logic/state_store/account_store.js]: There is no other property named `account` in `AccountStore`. So, I think it is fine. If you think it is really confusing then i can change it. :)

Comment (pablitovicente) [reply framework/src/modules/chain/logic/state_store/account_store.js]: No it's ok you have a good point :)

**Comment by SargeKhan** [comment]
@shuse2, the PR for `account.get` was not merged in the feature branch. So, I just cherry-picked the commits.

**Pull request merged**